.github/workflows/github-actions-unit-test.yml

```diff
@@ -67,16 +67,52 @@ jobs:
       - name: Unit Test NeuroId (ReactNative)
         run: ./gradlew :NeuroID:testReactNativeLibDebugUnitTest
 
-        # Execute unit tests
-      - name: Unit Test
+  android_sdk_integration_test_layout_app:
+    runs-on: [macos-12]
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+
+      - name: set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: 17
+          distribution: 'temurin'
+
+      # Execute integration sample tests
+      - name: Integration Test Sample app (compose)
+        uses: ReactiveCircus/android-emulator-runner@v2
+        with:
+          api-level: 29
+          profile: Nexus 6P
+          cores: 4
+          ram-size: 4096M
+          heap-size: 2048M
+          emulator-options: -no-snapshot-save -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
+          disable-animations: true
+          script: ./gradlew :testapp:connectedandroidLibDebugAndroidTest
+
+  android_sdk_integration_test_sample_app:
+    runs-on: [ macos-12 ]
```

Comment:
same here

Reply:
According to this article, https://medium.com/@esthcarelle/continuous-delivery-for-android-using-github-actions-effortless-android-testing-building-and-d3759ce66895, the reactivecircus emulator uses hardware acceleration only on MacOS which seems to make the tests more reliable which is why I moved these to separate sections to run the integration tests in MacOS. With ubuntu, this emulator seems to start up more times then not causing the tests to fail out with the "cannot find root window...." error.